packages/concerto-core/test/introspect/decorator.js

```diff
@@ -60,6 +61,24 @@ describe('Decorator', () => {
             d.isDecorator().should.equal(true);
 
         });
+        it('should throw an error for invalid type reference', () => {
+            const parent = null; // Placeholder, you need to provide a valid parent object
+            const ast = {
+                name: 'TestDecorator',
+                arguments: [
+                    {
+                        $class: `${MetaModelNamespace}.DecoratorTypeReference`,
+                        type: {
+                            name: 'NonExistentType',
+                            isArray: false
+                        }
+                    }
+                ]
+            };
+
+            const errorMessage=`Type '${ast.arguments[0].type.name}' not found within the namespace '${MetaModelNamespace}'. `;
+            (() => new Decorator(parent, ast)).should.throw(IllegalModelException,errorMessage );
+        });
```

Comment:
Is there an existing test that proves the happy path behaviour? If not, please add a test for the success case too.

Reply:
ok, I'll update it to do so